### PR TITLE
Fix url rewrite and draft of homepage

### LINKED ISSUE #6: Develop book review list view
Card link: https://github.com/codeday-labs/bookworms/projects/1#card-64509461

codeday-project-wireframe-list

In this view, an user can view all the book reviews.

### LINKED ISSUE #8: Unexpected url rewrite after clicking on `Add Review`
This issue happens when deployed to github pages.

**Steps to reproduce**
- Go to https://codeday-labs.github.io/bookworms/
- Click on `Add Review` button
- URL becomes `https://codeday-labs.github.io/create`
- Refresh the page and 404 shows up

**Expected result**
- The overwritten URL should be `https://codeday-labs.github.io/bookworms/create` instead
- The page should work after refreshing

### PR DESCRIPTION
**Pull request by dknv**
fixed URL rewrite when switching between 2 pages on remote env. - used basename in BrowserRouter (https://reactrouter.com/web/api/BrowserRouter/basename-string)

### CODE REVIEW

File: client/src/ReviewList.js [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+            {reviews.map((r) => (
+                <div className="review-preview" key={r.id}>
+                    <h1> {r.bookname} </h1>
+                    <h2> Published by {r.name} </h2>
+                    <p> {r.body} </p>
+                    <div className="footer">
+                        <p> {r.categories} </p>
+                        {!fav && <FavoriteBorderIcon className="footer-right" onClick={() => {favorite();}}/>}
+                        {fav && <FavoriteIcon className="footer-right" onClick={() => {favorite();}}/>}
+                    </div>
```
Comment (jeromewu) [inline client/src/ReviewList.js:27]: A better way to write this component:

```javascript
// destruct reviews from props directly
function ReviewList ({ reviews }) {

    const [fav, setFav] = useState(false);
   // func name start with a verb
    const toggleFavorite = () => {
      setFav(!fav)
    };

   // select component using shorthand if else
   const FavIcon = fav ? FavoriteIcon : FavoriteBorderIcon;

    return (
        <div className="review-list">
            {reviews.map((r) => (
                <div className="review-preview" key={r.id}>
                    <h1> {r.bookname} </h1>
                    <h2> Published by {r.name} </h2>
                    <p> {r.body} </p>
                    <div className="footer">
                        <p> {r.categories} </p>
                        <FavIcon
                          className="footer-right"
                          onClick={toggleFavorite}
                        />
                    </div>
                </div>
            ))}
        </div>
    );
}
```

File: client/src/favoriteButton.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const text = this.state.liked ? 'Like' : 'Unlike'
+    
+    return (
+      <button onClick={ this.handleClick }>{ text }</button>
+    );
+  }
+}
+
+
+ReactDOM.render(<LikeButton />, document.getElementById('root'));
```
Comment (jeromewu) [inline client/src/favoriteButton.js]: Should export this component instead of render.

File: client/src/favoriteButton.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';
+import FavoriteIcon from '@material-ui/icons/Favorite'; 
+
+class LikeButton extends React.Component {
```
Comment (jeromewu) [inline client/src/favoriteButton.js]: Better to write this as functional component instead of class component to be consistent.

**Review by jeromewu — CHANGES_REQUESTED** [review]
Overall comments:

1. Next time I think it is better to create a new branch to fix url rewrite instead of mixing it with `homepage-feed` branch, it makes this PR more clear and easy to merge.
2. Instead of putting all the CSS style in `index.css`, it is more recommended to use a css module, you can check more details here: [Adding Stylesheets](https://create-react-app.dev/docs/adding-a-stylesheet) and [BEM](https://medium.com/seek-blog/block-element-modifying-your-javascript-components-d7f99fcab52b)
3. In common practice, it is not recommended to commit .DS_Store file, please help to exclude / delete them using gitignore, ref: https://stackoverflow.com/questions/18393498/gitignore-all-the-ds-store-files-in-every-folder-and-subfolder

**Comment by dknv** [comment]
I tried remove DS.Store in all directories manually via terminal and it seems to disappear on the branches, so hopefully, it will ignore DS.Store from now on when I commit something...!

**Comment by dknv** [comment]
I am still trying out how and where to display the filter for category [still currently working on it] -- but the homepage version that was pushed as of now is an okay version (not 100% functional but the idea is there). Would you mind look it over and see if it works for now? @jeromewu

**Comment by jeromewu** [comment]
OK, let's not to include filter in this PR to be more agile, I will start the code review now.

File: client/src/tools/SearchBar.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        setIsSearching(false);
+    };
+    
+    const BarIcon = isSearching ? CloseIcon : SearchIcon;
+
+    return (
+        <div className={styles.search}>
+            <input 
+                type="text" 
+                className={styles.inputArea && styles.searchTerm}
```
Comment (jeromewu) [inline client/src/tools/SearchBar.js]: For variable like `styles.inputArea`, the value is actually a string (you can console.log to check), so in JavaScript when you use AND operator with strings, it returns the last string:

```java
const s1 = 'foo';
const s2 = 'bar';

console.log(s1 && s2); // bar
```
Which I think is not the purpose of this line of code

**Review by jeromewu — CHANGES_REQUESTED** [review]
Instead of `tools/`, most of the time people name that folder `components/` to indicate that they are components that might be reused. It is recommended to use that.

Other than that, only 1 review comment left. :smile:

**Comment by dknv** [comment]
Sorry for the late notification!

**Review by jeromewu — APPROVED** [review]

**Comment by jeromewu** [comment]
Thanks for the hard work, it is merged now. :tada:

**Pull request merged**